src/osd/osd_types.h

```diff
@@ -1922,6 +1922,11 @@ struct pg_pool_t {
   /// choose a random hash position within a pg
   uint32_t get_random_pg_position(pg_t pgid, uint32_t seed) const;
 
+  /// EC partial writes: test if a shard is a non-primary
+  bool is_nonprimary_shard(const shard_id_t shard) const {
+    return !nonprimary_shards.empty() && nonprimary_shards.contains(shard);
```

Comment:
ACK, `is_nonprimary_shard` returns `false` for empty `nonprimary_shards` (legacy EC paths).